### PR TITLE
perf: improve performance of ZCAP capability creation

### PR DESCRIPTION
**Pull request by aholovko**
This PR adds support for context caching for LD document loader. This improves performance of onboarding flow by 30-40%.

part of #146

Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

File: cmd/kms-rest/startcmd/lddocument_loader.go [hunk 1]
```diff
@@ -140,6 +205,10 @@ func loadLDContext() (map[string]*ld.RemoteDocument, error) {
 			vocab:   "https://w3id.org/security/v2",
 			content: w3idOrgSecurityV2,
 		},
+		{
+			vocab:   "https://trustbloc.github.io/context/vc/credentials-v1.jsonld",
```
Comment (llorllale) [inline cmd/kms-rest/startcmd/lddocument_loader.go:209]: Where is this context referenced? Is it from aries-framework-go while adding `JsonWebSignature2020` proofs?

File: cmd/kms-rest/startcmd/start.go [hunk 2]
```diff
@@ -650,7 +650,10 @@ func startKmsService(params *kmsRestParameters, srv Server) error {
 
 	kmsRouter := router.PathPrefix(operation.KMSBasePath).Subrouter()
 
-	kmsREST := operation.New(config)
+	kmsREST, err := operation.New(config)
+	if err != nil {
+		return err
```
Comment (llorllale) [inline cmd/kms-rest/startcmd/start.go:655]: Please wrap this error.

Comment (aholovko) [reply cmd/kms-rest/startcmd/start.go]: Done

**Comment by llorllale** [comment]
I am also confused about this statement:

> This improves performance of onboarding flow by 30-40%.

This zcap is created once in the onboarding flow. There are a _bunch_ of other calls to the KMS during the flow. Are you saying that optimizing this _one_ step will speed up the onboarding flow E2E by 30-40%?

File: cmd/kms-rest/startcmd/lddocument_loader.go (hunk 1 above)
Comment (aholovko) [reply cmd/kms-rest/startcmd/lddocument_loader.go:209]: I think so, I got it from tracing network calls from loader, also this is what is listed [here](https://github.com/hyperledger/aries-framework-go/blob/main/pkg/doc/signature/proof/cache.go#L9)

**Comment by aholovko** [comment]
> I am also confused about this statement:
> 
> > This improves performance of onboarding flow by 30-40%.
> 
> This zcap is created once in the onboarding flow. There are a _bunch_ of other calls to the KMS during the flow. Are you saying that optimizing this _one_ step will speed up the onboarding flow E2E by 30-40%?

yep, I will share screenshots from tracing a bit later

File: cmd/kms-rest/startcmd/lddocument_loader.go (hunk 1 above)
Comment (llorllale) [reply cmd/kms-rest/startcmd/lddocument_loader.go:209]: I was going to say that aries-framework-go is _supposed_ to already be adding this context - apparently that is not working properly.

cc @kdimak

**Review by llorllale — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**